src/script/engine.zig

```diff
@@ -486,6 +487,20 @@ pub const Engine = struct {
         try self.stack.pushByteArray(&hash_160);
     }
 
+    /// OP_HASH256: The input is hashed two times with SHA-256.
+    fn opHash256(self: *Engine) EngineError!void {
+        const data = try self.stack.pop();
+        defer self.allocator.free(data);
+
+        var hash1: [Sha256.digest_length]u8 = undefined;
+        Sha256.hash(data, &hash1, .{});
+
+        var hash2: [Sha256.digest_length]u8 = undefined;
+        Sha256.hash(&hash1, &hash2, .{});
```

Comment:
```zig
    var digest: [Sha256.digest_length]u8 = undefined;
    Sha256.hash(data, &digest, .{});
    Sha256.hash(&digest, &digest, .{});
```

I believe this does compile and work. Please check it

Reply:
@tdelabro, it works! loved it